File: client/src/types/site.ts
/** Contrato de GET /api/site — espejo de Models/Site.cs. */
export interface Site {
  name: string
  slogan: string
  founded: string
  story: string[]
  values: { title: string; text: string }[]
  location: {
    address: string
    district: string
    reference: string
    directions: string[]
    mapQuery: string
    hours: { days: string; open: string; close: string }[]
  }
  contact: {
    phones: string[]
    whatsapp: string
    whatsappMessage: string
    instagram: string
    facebook: string
    tiktok: string
  }
}

File: src/CaballeroNegro.Api/Models/Site.cs
namespace CaballeroNegro.Api.Models;

/// <summary>Datos del restaurante (historia, ubicación, contacto). Fuente: Data/site.json.</summary>
public sealed record Site(
    string Name,
    string Slogan,
    string Founded,
    IReadOnlyList<string> Story,
    IReadOnlyList<SiteValue> Values,
    Location Location,
    Contact Contact);

public sealed record SiteValue(string Title, string Text);

public sealed record Location(
    string Address,
    string District,
    string Reference,
    IReadOnlyList<string> Directions,
    string MapQuery,
    IReadOnlyList<OpeningHours> Hours);

public sealed record OpeningHours(string Days, string Open, string Close);

public sealed record Contact(
    IReadOnlyList<string> Phones,
    string Whatsapp,
    string WhatsappMessage,
    string Instagram,
    string Facebook,
    string Tiktok);
